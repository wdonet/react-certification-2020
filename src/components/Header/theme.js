const base = {
  easeOutBack: 'cubic-bezier(0.34, 1.56, 0.64, 1)',
  colorWhite: 'rgb(255, 255, 255)',
  colorBlack: 'rgb(0, 0, 0)',
};

const dark = {
  id: 'dark',
  ...base,
  backgroundColor: '#333',
  textColor: 'black',
};

const light = {
  id: 'light',
  ...base,
  backgroundColor: '#666',
  textColor: 'white',
};

export const theme = { dark, light };
