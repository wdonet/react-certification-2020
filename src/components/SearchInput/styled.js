import styled, { css } from 'styled-components';

const SearchInputContainer = styled.div`
  cursor: text;
  display: inline-flex;
  align-items: center;
  font-weight: 400;
  font-style: normal;
  color: ${({ theme }) => theme.font.primary};
  appearance: none;
  border: 1px solid ${({ theme }) => theme.background.tertiary};
  border-radius: 0;
  width: 100%;
  font-size: 14px;
  line-height: 27px;
  background-color: ${({ theme }) => theme.background.primary};
  box-sizing: border-box;

  ${({ theme, focused }) =>
    focused &&
    css`
      outline: none;
      background: ${theme.background.secondary};
      border: 1px solid ${theme.background.highlight};
    `}

  width: 100%;
  @media (max-width: 900px) {
    height: 40px;
  }
  @media (max-width: 700px) {
    height: 30px;
  }
  @media (max-width: 400px) {
    height: 20px;
  }
`;

const StyledInput = styled.input`
  letter-spacing: inherit;
  color: currentColor;
  padding: 0 10px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 45px;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  &:focus {
    outline: 0;
  }
`;

const InputAdornment = styled.div`
  cursor: pointer;
  display: inline-flex;
  align-items: center;
  > * {
    margin-right: 10px;
  }
`;

export { StyledInput, SearchInputContainer, InputAdornment };
