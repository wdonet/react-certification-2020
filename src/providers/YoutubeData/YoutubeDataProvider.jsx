import React, { useEffect, useReducer } from 'react';
import { getVideosByQuery, getVideoDetails, getRelatedVideos } from './api';
import reducer from './reducer';
import { setSearchTerm, setVideos } from './actions';

const YoutubeDataContext = React.createContext(null);

const YoutubeDataProvider = ({ iframeAPIReady, children }) => {
  const [state, dispatch] = useReducer(reducer, {
    videos: [],
    searchTerm: '',
  });

  const { videos, searchTerm } = state;

  const fetchVideos = async (query) => {
    try {
      const newVideos = await getVideosByQuery(query);
      setVideos(dispatch, newVideos);
    } catch (error) {
      console.error(error);
    }
  };

  const fetchVideoDetails = async (videoId) => {
    try {
      const videoDetails = await getVideoDetails(videoId);
      return videoDetails;
    } catch (error) {
      console.error(error);
    }
  };

  const fetchRelatedVideos = async (videoId) => {
    try {
      const relatedVideos = await getRelatedVideos(videoId);
      return relatedVideos;
    } catch (error) {
      console.error(error);
    }
  };

  useEffect(() => {
    fetchVideos();
  }, []);

  return (
    <YoutubeDataContext.Provider
      value={{
        videos,
        fetchVideos,
        iframeAPIReady,
        fetchVideoDetails,
        fetchRelatedVideos,
        searchTerm,
        setSearchTerm: (search) => setSearchTerm(dispatch, search),
      }}
    >
      {children}
    </YoutubeDataContext.Provider>
  );
};

YoutubeDataProvider.propTypes = {};

export { YoutubeDataContext };
export default YoutubeDataProvider;
