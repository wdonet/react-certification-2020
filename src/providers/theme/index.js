export { default, useTheme } from './Theme.provider';
