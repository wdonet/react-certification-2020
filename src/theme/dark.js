const dark = {
  palette: {
    primary: {
      main: '#556cd6',
      transparent: '3,45,70,0.4',
    },
    secondary: {
      main: '#303030',
      light: '#424242',
    },
    accent: '#19857b',
  },
  typography: {
    title: {
      color: '#fff',
    },
    text: {
      color: '#fafafa',
    },
  },
  appBar: {
    height: {
      desktop: '64px',
    },
  },
  profile: {
    primary: '#303030',
    secondary: '#636363',
  },
};

export default dark;
