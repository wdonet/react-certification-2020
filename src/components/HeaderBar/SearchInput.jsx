import React, { useState } from 'react';
import styled from 'styled-components';

const Input = styled.input`
  height: 40px;
  font-size: 25px;
  width: 100%;
  border: 2px solid #aaa;
  border-radius: 4px;
  margin: 8px 0;
  outline: none;
  padding: 8px;
  box-sizing: border-box;
  transition: 0.3s;
  padding-left: 50px;
  cursor: pointer;

  &:focus {
    border-color: dodgerBlue;
    box-shadow: 0 0 8px 0 dodgerBlue;
  }

  :focus + .left-icon {
    svg {
      fill: dodgerBlue;
    }
  }
`;
const StyledInput = styled.div`
  &.inputWithIcon {
    position: relative;
    width: 250px;
  }

  .left-icon {
    position: absolute;
    left: 5px;
    top: 50%;
    transform: translateY(-50%);
    svg {
      fill: black;
      transition: 0.3s;
    }
  }

  button.right-icon {
    background: none;
    border: none;
    position: absolute;
    right: 5px;
    top: 50%;
    transform: translateY(-50%);
    svg {
      fill: black;
      transition: 0.3s;
    }
  }
  @media (max-width: 40em) {
    width: 100%;
  }
`;

function SearchInput({ placeholder, updateQuery }) {
  const [text, setText] = useState('');

  const handleChange = (event) => {
    setText(event.target.value);

  };

  const handleSubmit = (event) => {
    updateQuery(text);
    event.preventDefault();
  }
  const clearInput = () => {
    setText('');
    if (text !== '') {
      updateQuery('');
    }
  };

  return (
    <form  onSubmit={handleSubmit}>
      <StyledInput className="inputWithIcon">
        <Input
          type="text"
          value={text}
          onChange={handleChange}
         
          placeholder={placeholder}
        />
        <div className="left-icon">
          <svg
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24 24"
            width="30px"
            height="30px"
          >
            <path d="M 13.261719 14.867188 L 15.742188 17.347656 C 15.363281 18.070313 15.324219 18.789063 15.722656 19.1875 L 20.25 23.714844 C 20.820313 24.285156 22.0625 23.972656 23.015625 23.015625 C 23.972656 22.058594 24.285156 20.820313 23.714844 20.25 L 19.191406 15.722656 C 18.789063 15.324219 18.070313 15.363281 17.347656 15.738281 L 14.867188 13.261719 Z M 8.5 0 C 3.804688 0 0 3.804688 0 8.5 C 0 13.195313 3.804688 17 8.5 17 C 13.195313 17 17 13.195313 17 8.5 C 17 3.804688 13.195313 0 8.5 0 Z M 8.5 15 C 4.910156 15 2 12.089844 2 8.5 C 2 4.910156 4.910156 2 8.5 2 C 12.089844 2 15 4.910156 15 8.5 C 15 12.089844 12.089844 15 8.5 15 Z" />
          </svg>
        </div>

        <button type="button" className="right-icon" onClick={clearInput}>
          <svg width="24pt" height="24pt" viewBox="0 0 24 24" version="1.1">
            <g id="surface1">
              <path d="M 12 1.546875 C 6.203125 1.546875 1.5 6.25 1.5 12.046875 C 1.5 17.84375 6.203125 22.546875 12 22.546875 C 17.796875 22.546875 22.5 17.84375 22.5 12.046875 C 22.5 6.25 17.796875 1.546875 12 1.546875 Z M 17.078125 15.585938 C 17.148438 15.65625 17.183594 15.75 17.183594 15.847656 C 17.183594 15.945312 17.148438 16.046875 17.078125 16.109375 L 16.0625 17.128906 C 15.988281 17.203125 15.894531 17.234375 15.800781 17.234375 C 15.707031 17.234375 15.609375 17.199219 15.539062 17.128906 L 12 13.585938 L 8.464844 17.132812 C 8.394531 17.207031 8.296875 17.242188 8.203125 17.242188 C 8.109375 17.242188 8.011719 17.203125 7.941406 17.132812 L 6.929688 16.117188 C 6.859375 16.046875 6.820312 15.953125 6.820312 15.851562 C 6.820312 15.753906 6.859375 15.65625 6.929688 15.589844 L 10.476562 12.027344 L 6.917969 8.511719 C 6.773438 8.367188 6.773438 8.128906 6.917969 7.984375 L 7.929688 6.964844 C 8 6.894531 8.09375 6.859375 8.195312 6.859375 C 8.292969 6.859375 8.386719 6.894531 8.457031 6.964844 L 12.003906 10.46875 L 15.554688 6.964844 C 15.625 6.894531 15.71875 6.859375 15.816406 6.859375 C 15.914062 6.859375 16.007812 6.894531 16.078125 6.964844 L 17.089844 7.984375 C 17.234375 8.128906 17.234375 8.367188 17.089844 8.511719 L 13.53125 12.027344 Z M 17.078125 15.585938 " />
            </g>
          </svg>
        </button>
      </StyledInput>
    </form>
  );
}

export default SearchInput;
