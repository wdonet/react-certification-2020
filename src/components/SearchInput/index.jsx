import React, { useState } from 'react';
import { faSearch } from '@fortawesome/free-solid-svg-icons';
import { StyledInput, SearchInputContainer, InputAdornment } from './styled';
import { useYoutubeData } from '../../providers/YoutubeData';
import Icon from '../Icon';

const SearchInput = () => {
  const [search, setSearch] = useState('');
  const [focused, setFocused] = useState(false);

  const { fetchVideos } = useYoutubeData();

  const handleChange = (e) => {
    const {
      target: { value },
    } = e;

    setSearch(value);
  };

  const handleKeyDown = (e) => {
    if (e.key === 'Enter') {
      fetchVideos(search);
    }
  };
  return (
    <SearchInputContainer focused={focused}>
      <StyledInput
        placeHolder="Search"
        onChange={handleChange}
        onKeyDown={handleKeyDown}
        onFocus={() => setFocused(true)}
        onBlur={() => setFocused(false)}
      />
      <InputAdornment onClick={() => fetchVideos(search)}>
        <Icon icon={faSearch} size="xsmall" />
      </InputAdornment>
    </SearchInputContainer>
  );
};

export default SearchInput;
