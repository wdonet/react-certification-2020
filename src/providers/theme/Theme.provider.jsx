// React
import React, { useState, useEffect, useContext, useCallback } from 'react';
// Utils
import { THEME_SELECTED } from '../../utils/constants';
import { storage } from '../../utils/storage';
// Themes
import light from '../../theme/light';
import dark from '../../theme/dark';

const ThemeContext = React.createContext({ theme: light, selected: false });

const useTheme = () => {
  const context = useContext(ThemeContext);
  if (!context) {
    throw new Error(`Can't use "useTheme" without an ThemeProvider!`);
  }
  return context;
};

const ThemeBaseProvider = ({ children }) => {
  const [theme, setTheme] = useState(light);
  const [selected, setSelected] = useState(false);

  useEffect(() => {
    const lastTheme = storage.get(THEME_SELECTED);
    const whichTheme = lastTheme ? dark : light;
    setSelected(lastTheme);
    setTheme(whichTheme);
  }, []);

  const toggleTheme = useCallback((toggle) => {
    setTheme(toggle ? dark : light);
    storage.set(THEME_SELECTED, toggle);
  }, []);

  return (
    <ThemeContext.Provider value={{ theme, selected, toggleTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

export { useTheme, ThemeContext };
export default ThemeBaseProvider;
