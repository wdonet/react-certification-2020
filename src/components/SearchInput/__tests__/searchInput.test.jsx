import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';

import SearchInput from '../index';
import YoutubeDataProvider from '../../../providers/YoutubeData';

import { lightTheme } from '../../../theme';

describe('SearchInput', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <YoutubeDataProvider>
          <ThemeProvider theme={lightTheme}>
            <SearchInput />
          </ThemeProvider>
        </YoutubeDataProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
