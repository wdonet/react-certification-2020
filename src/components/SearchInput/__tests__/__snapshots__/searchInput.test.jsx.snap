// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders correctly 1`] = `
.c0 {
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 400;
  font-style: normal;
  color: #333333;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #f0dddd85;
  border-radius: 0;
  width: 100%;
  font-size: 14px;
  line-height: 27px;
  background-color: #ECECEC;
  box-sizing: border-box;
  width: 100%;
}

.c1 {
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 0 10px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 45px;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
}

.c1:focus {
  outline: 0;
}

.c2 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 > * {
  margin-right: 10px;
}

@media (max-width:900px) {
  .c0 {
    height: 40px;
  }
}

@media (max-width:700px) {
  .c0 {
    height: 30px;
  }
}

@media (max-width:400px) {
  .c0 {
    height: 20px;
  }
}

<div
  className="c0"
>
  <input
    className="c1"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
  <div
    className="c2"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-search fa-w-16 fa-1x "
      data-icon="search"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
